### PR TITLE
New Security Toys

### PR DESCRIPTION
**Pull request by Kurfursten**
**Battering Ram**
- Spawns in the Warden's locker, used in raids
- Bulky. Musttwohand and too large for a backpack.
- Breaks down unbolted, unwelded airlocks at the same rate as a drill through a wall (6 seconds) - leaves behind an unanchored assembly and blown out airlock electronics. Very messy - for authorized use only.
- If bolted, takes the same amount of time as drilling an R-wall (25 seconds) 
- Add an additional 3 seconds to both cases if welded
- Does not affect blast doors or walls at all
- 15 force

**Secure GPS**
- Spawns in the Security Officer locker
- Displays only other Secure GPS
- Secure GPS do not appear on other GPS
- If you drop a secure GPS (or it is stripped from you), it displays a warning message with your location to all other secure GPS
- If you die while wearing (on belt, pockets, etc. -- not in backpack) the device, it displays a warning message with your location to all other secure GPS

---

Do not merge until a sprite is added for the battering ram. Currently using a placeholder.

### CODE REVIEW

File: code/modules/telesci/gps.dm [hunk 1]
```diff
@@ -101,3 +111,31 @@ var/list/GPS_list = list()
...
+	..()
+	for(var/obj/item/device/gps/secure/S in SPS_list)
+		S.announce(wearer, src, "died")
+
+/obj/item/device/gps/secure/dropped(mob/wearer as mob)
+	..()
+	if(!istype(src.loc, /turf))
+		return
+	for(var/obj/item/device/gps/secure/S in SPS_list)
+		S.announce(wearer, src, "lost his SPS")
```
Comment (ComicIronic) [inline code/modules/telesci/gps.dm]: >his

I suggest 
`"lost [wearer.gender == FEMALE ? "her" : "his"] "SPS"`

**Comment by Probe1** [comment]
The battering ram is a very smart addition.  Its in line with diamond drills essentially, and also secure as well as limited.  Good design.

The gps seems okay at first glance but potentially powerful.  Where is this warning message displayed?

**Comment by ComicIronic** [comment]
I'd love it if the battering ram made the door assembly fly back into the room.

**Comment by FudgePucker** [comment]
Maybe being able to hack/emag (hack would be preferable, emag shouldn't be some fucking magical do all device) to send back a false 'Everything is fine and I am totally not dead and spaced' report on the GPS.'

**Comment by Nienhaus** [comment]
dssucx1
That sprite though.

**Comment by Nienhaus** [comment]
@Kurfursten http://puu.sh/hyRUO/c217235078.dmi This looks a bit more like a battering ram.

**Comment by clusterfack** [comment]
>I'd love it if the battering ram made the door assembly fly back into the room.

Holy shit this, also it would be pretty easy to code as well

File: code/modules/telesci/gps.dm (hunk 1 above)
Comment (Nienhaus) [inline code/modules/telesci/gps.dm]: Or just change it to \his
Pretty sure that'd work.

**Comment by PJB3005** [comment]
And then have it able to knock people unconsious too.

Due process: SS13 edition.

**Comment by Exxion** [comment]
Actual clip of security with this:

image

**Comment by Kurfursten** [comment]
@Nienhaus Having difficulty with that sprite. Could you try uploading somewhere else?

File: code/modules/telesci/gps.dm (hunk 1 above)
Comment (BestName) [inline code/modules/telesci/gps.dm]: Or just change it to "their"

**Comment by FudgePucker** [comment]
Oh shit I just came up with the best fucking idea ever.
Kill an officer, then put a polyacid grenade with a proxy sensor on it.
They go locate the body, then they die in a cloud of death.

**Comment by Nienhaus** [comment]
@Kurfursten you should be able to right click the page and click save as.
If not, 
a

**Comment by SkowronX** [comment]
Thats some weird perspective.

**Comment by Nienhaus** [comment]
kayo_1man
It was based off that.

**Comment by mrpain666** [comment]
How would a battering ram interact with a regular, unbolted, working door that I don't have access too? Say I'm responding to a cult call in science, and I dont have access?

**Comment by Kurfursten** [comment]
@mrpain666 6 seconds

**Comment by mrpain666** [comment]
Oh, so it would actually open the door then? Neat.

**Comment by Exxion** [comment]
No, it would break it. Breaking doors down is its purpose.

**Comment by Kurfursten** [comment]
@ComicIronic Update tags

**Comment by FudgePucker** [comment]
Cargos gone independent, about to order 100 more eguns and shotguns.
When all of a sudden there is a loud banging noise outside the door.
And then a emag'd hailer goes off.

Fuckin shitcurity to a whole new level.

**Comment by FudgePucker** [comment]
>Now throws electronics

Okay if this sounds as fucking awesome as it sounds, you need to fuckin give us a webm to jack off to.

**Comment by Kurfursten** [comment]
@FudgePucker It kicks it out pretty far -- as far as 10 squares at a decent speed. However, because airlock electronics have such low throw force, it only deals about .5 brute.

I'll put up a webm if I have time later, but exams are closing in, and it might get merged soon anyway.

**Comment by Midaychi** [comment]
Just making sure, but you sanity checked it right? What happens if someone initiates a ramming sequence and the ai or a traitor with an emag or an innocent unknowing drone changes the door's state mid-ram?  Ie, Ai unbolts and opens a door, or traitor welds and emags the door to be a dick?

**Comment by Probe1** [comment]
The idea of a critical failure and someone emagging it sending the door skyrocketing towards them sounds way fucking awesome.

**Comment by Kurfursten** [comment]
If someone opens the door mid ram the door will still be smashed if the rammer doesn't move

I'll do some more testing

**Pull request merged**